cumulusci/tasks/salesforce/sourcetracking.py

```diff
@@ -128,8 +130,15 @@ def _filter_changes(self, changes):
             ) or any(re.search(s, full_name) for s in self._exclude):
                 ignored.append(change)
             else:
-                filtered.append(change)
-        return filtered, ignored
+                if not separate_profiles and mdtype == "Profile":
+                    profiles.append(name)
+                else:
+                    filtered.append(change)
+        return (
+            (filtered, ignored, profiles)
+            if not separate_profiles
+            else (filtered, ignored)
+        )
```

Comment:
❓: Have you considered keeping the previous behavior of this function the same? Right now, the method can return two different types based on the `separate_profiles` parameter. This can be confusing.

It's best if a method always returns the same type. This makes it easier to understand and use. If a method returns different types, it can cause errors or need extra checks when we use it.

What about moving this logic to `retrieve_changes`?

Reply:
Sure I've moved it into a separate function